src/main/scala/com/vngrs/etl/Pipeline.scala

```diff
@@ -0,0 +1,94 @@
+package com.vngrs.etl
+
+import com.vngrs.etl.combiners.{Union, Zip}
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+
+import scala.reflect.ClassTag
+
+/**
+  * Represents ETL Pipeline.
+  *
+  * @param rdd The [[org.apache.spark.rdd.RDD]] which will be ETLed
+  * @tparam A Type of the [[org.apache.spark.rdd.RDD]]
+  */
+final class Pipeline[A: ClassTag](val rdd: RDD[A]) {
```

Comment:
`Pipeline` class makes `Extract` trait useless. Extract could be a factory that creates `Pipeline` and implemented as an object.